quantization/src/encoded_vectors_binary.rs

```diff
@@ -33,6 +36,8 @@ impl<TStorage: EncodedStorage> EncodedVectorsBin<TStorage> {
     ) -> Result<Self, EncodingError> {
         debug_assert!(validate_vector_parameters(orig_data.clone(), vector_parameters).is_ok());
 
+        let (alpha, _) = Self::find_alpha_offset_size_dim(orig_data.clone());
```

Comment:
I have not used the offset (min value of the dataset) here, but I think it would make much sense to use it when encoding the vectors. It could widen the compatibility range of vector extractor models for binary quantization.

It would also make the implementation very symmetrical with UINT8 quantization, with the only difference between them being the quantization ratio and the actual similarity calculation.

Reply:
I got confused before; in the case of binary vectors, offset should not be the min value but the mean of the dataset to split the data in half.

Ideally it should be an offset vector not a single f32. It's a whole other set of experiments to be run on a different branch :)